lib/streamlit/runtime/state/query_params_proxy.py

```diff
@@ -205,3 +207,11 @@ def from_dict(self, params):
         """
         with get_session_state().query_params() as qp:
             return qp.from_dict(params)
+
+    @staticmethod
+    def missing_key_error_message(key: str) -> str:
+        return f'st.query_params has no key "{key}".'
```

Comment:
nit: an alternative could also be: "There is no query parameter with the name {key}" for both situations. Maybe do quick check with @sfc-gh-dmatthews and/or @jrieke about the preferred wording.